bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java

```diff
@@ -0,0 +1,356 @@
+                value = (OnOffType)command == OnOffType.ON ? "1" : "0";
+                break;
+            default:
+                break;
+        }
+        if (key != null) {
+            sendData(key, value);
+        } else {
+            logger.warn("Could not update channel {} because it is read only", channelUID.getId());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(GoEChargerConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        startAutomaticRefresh();
+        logger.debug("Finished initializing!");
+    }
+
+    private String getUrl(String type) {
+        String urlStr = type;
+        return urlStr.replace("%IP%", StringUtils.trimToEmpty(config.ip));
+    }
+
+    private void sendData(String key, String value) {
+        String errorMsg = null;
+        String urlStr = getUrl(GoEChargerBindingConstants.MQTT_URL).replace("%KEY%", key).replace("%VALUE%", value);
+        logger.debug("POST URL = {}", urlStr);
+
+        String result = null;
+        try {
+            ContentResponse contentResponse = httpClient.newRequest(urlStr).method(HttpMethod.POST)
+                    .timeout(5, TimeUnit.SECONDS).send();
+
+            result = contentResponse.getContentAsString();
+            // TODO check if value really changed
+            logger.debug("Response: {}", result);
+        } catch (Exception e) {
+            errorMsg = e.getMessage();
+        }
+
+        if (result != null) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+            return;
+        }
+
+        logger.warn("Error in Go-eCharger request: {}", errorMsg);
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, errorMsg);
+    }
+
+    /**
+     * Request new data from Go-E charger
+     *
+     * @return the Go-E charger object mapping the JSON response or null in case of
+     *         error
+     */
+    @Nullable
+    private GoEStatusResponse getGoEData() {
+        GoEStatusResponse result = null;
+        String errorMsg = null;
+
+        String urlStr = getUrl(GoEChargerBindingConstants.API_URL);
+        logger.debug("GET URL = {}", urlStr);
+
+        try {
+            ContentResponse contentResponse = httpClient.newRequest(urlStr).method(HttpMethod.GET)
+                    .timeout(5, TimeUnit.SECONDS).send();
+
+            result = gson.fromJson(contentResponse.getContentAsString(), GoEStatusResponse.class);
+            // logger.debug("Response: {}", contentResponse.getContentAsString());
+        } catch (Exception e) {
+            errorMsg = e.getMessage();
+        }
+
+        if (result != null) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+            return result;
+        }
+
+        retryCounter++;
+        if (retryCounter == 1) {
+            logger.warn("Error in getting data from Go-E charger, retrying once");
+            return getGoEData();
+        }
```

Comment:
Why would you need to retry immediately? Your refreshJob does retries just by virtue that it is periodic.

Reply:
The idea is to not set null values immediately but try again in case there was just a short connection issue. So only after the next retry values are set to null/undef. Took this overfrom fronius boinding. I guess the values are a bit more stable like this in shaky wifi connections.